app/src/main/java/org/dhis2/usescases/main/program/ProgramPresenter.kt

```diff
@@ -99,110 +79,42 @@ class ProgramPresenter internal constructor(
         )
     }
 
-    override fun dispose() {
-        compositeDisposable!!.clear()
+    fun onSyncStatusClick(program: ProgramViewModel) {
+        view.showSyncDialog(program)
     }
 
-    override fun onSyncStatusClick(program: ProgramViewModel) {
-        view!!.showSyncDialog(
-            SyncStatusDialog.Builder()
-                .setConflictType(
-                    if (program.typeName() != "DataSets") {
-                        SyncStatusDialog.ConflictType.PROGRAM
-                    } else {
-                        SyncStatusDialog.ConflictType.DATA_SET
```

Comment:
For next time, SyncStatusDialog can actually be part of the program view model and use as an attribute when initialization of the dialog. If not, pass it as parameter. At the we we'd like to be the view as dumb as possible